src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -8008,47 +7981,42 @@ export class StudyViewPageStore
     readonly molecularProfileSampleCounts = remoteData<
         MultiSelectionTableRow[]
     >({
-        await: () => [this.molecularProfiles, this.selectedSamples],
+        await: () => [this.molecularProfiles],
         invoke: async () => {
-            // return empty if there are no filtered samples
-            if (!this.hasFilteredSamples) {
-                return [];
-            }
-            const counts = await internalClient.fetchMolecularProfileSampleCountsUsingPOST(
-                {
+            const [counts, selectedSamples] = await Promise.all([
+                internalClient.fetchMolecularProfileSampleCountsUsingPOST({
                     studyViewFilter: this.filters,
-                }
-            );
+                }),
+                toPromise(this.selectedSamples),
+            ]);
 
             return counts.map(caseListOption => {
                 return {
                     uniqueKey: caseListOption.value,
                     label: caseListOption.label,
                     numberOfAlteredCases: caseListOption.count,
-                    numberOfProfiledCases: this.selectedSamples.result.length,
+                    numberOfProfiledCases: selectedSamples.length,
                 } as any;
             });
         },
         default: [],
     });
 
     readonly caseListSampleCounts = remoteData<MultiSelectionTableRow[]>({
-        await: () => [this.selectedSamples],
         invoke: async () => {
-            // return empty if there are no filtered samples
-            if (!this.hasFilteredSamples) {
-                return [];
-            }
-            const counts = await internalClient.fetchCaseListCountsUsingPOST({
-                studyViewFilter: this.filters,
-            });
+            const [counts, selectedSamples] = await Promise.all([
+                internalClient.fetchCaseListCountsUsingPOST({
+                    studyViewFilter: this.filters,
+                }),
+                toPromise(this.selectedSamples),
```

Comment:
here we're taking selectedSamples out of await, and awaiting it in the body of invoke